src/inspector_agent.cc

```diff
@@ -779,7 +797,8 @@ void Agent::WaitForDisconnect() {
   }
   // TODO(addaleax): Maybe this should use an at-exit hook for the Environment
   // or something similar?
-  client_->contextDestroyed(parent_env_->context());
+  if (!client_->reportWaitingForDebuggerToDisconnect())
```

Comment:
Maybe put this to NodeInspectorClient::contextDestroyed?

Reply:
I'd like to have it here, otherwise in future someone might use contextDestroyed in other places and get misleading notification.

Reply:
Feel free to rename "contextDestroyed" to something else - like "mainContextDone"?